specification/attestation.adoc

```diff
@@ -1,309 +1,886 @@
+++++
+
+Asymmetric key pairs can be derived from a CDI in order to generate the
+attestation key for a DICE layer. This key can then be used to attest of the
+next TCB layer.
+
+:stem: asciimath
+[.center]
+[stem]
+++++
+(UDS\_PrivKey,\ UDS\_PubKey) = AsymKDF(UDS)
+++++
+:stem: asciimath
+[.center]
+[stem]
+++++
+(CDI\_PrivKey_{N},\ CDI\_PubKey_{N}) = AsymKDF(CDI_{N})
+++++
+
+A fixed length TCB layer identifier called `CDI_ID` can be derived from its
+attestation key:
+
+:stem: asciimath
+[.center]
+[stem]
+++++
+UDS\_ID = KeyDerivationFunction(ID_{Len}, UDS\_PubKey)
+++++
+:stem: asciimath
+[.center]
+[stem]
+++++
+CDI\_ID_{N} = KeyDerivationFunction(ID_{Len}, CDI\_PubKey_{N})
+++++
+
+When loading the next TCB layer, the attesting layer provides it with both its
+CDI and its attestation certificate.
+
+The TVM guest can obtain an Attestation Evidence from the TSM by calling the
+TG-ABI `sbi_tee_guest_get_evidence()` FID. The guest provides a challenge value
+to the TSM through this call, and that value must be included in the generated
+Evidence. This value allows relying parties to establish the Attestation
+Evidence freshness.
+
+The Evidence is signed by the TSM attestation key, which is HW RoT-rooted.
+A Relying Party can then verify and authenticate the Evidence with platform
+owner or manufacturer provided Endorsements, like e.g. a Trust Anchor.
+
+The TCB extension and evidence collection for a TVM attestation is shown
+below:
 
-* Create TVM attestation structure in a temporary buffer in per-hart 
-confidential memory
-* Populate TVM TcbEvidenceInfo per the TVM measurements recorded by the TSM
-* Copy additional data from CSR <TBD>
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "Layered Attestation architecture for TVMs"]
+image::layered_attestation.svg[]
+
+It is expected that an implementation will provide implementation-specific
+intrinsics to record measurements of the TSM into the firmware RoT for
+measurement to support the AP-TEE layered RTMs attestation of AP-TEE workloads.
+
+==== Evidence Format
+
+The AP-TEE Attestation Evidence uses the IETF Entity Attestation Token
+(<<EAT>>), formatted as an untagged Concise Binary Object Representation
+(<<CBOR>>) Web Token (<<CWT>>).  An AP-TEE EAT profile is proposed to narrow
+the EAT specification for the AP-TEE use case to enable interoperability.
+
+The CWT is composed of one EAT submodule Claims-Set map where the map values
+are nested tokens for the TVM, TSM and Platform Claims.
+
+The TVM EAT is a CWT tagged CBOR formatted token, wrapped with a
+COSE_Sign1 <<COSE>> envelope. It is signed by the TSM attestation key and must
+contain a TVM provided challenge, e.g. a Relying Party provided nonce for
+establishing Evidence freshness.
+
+The TSM EAT is a CWT tagged CBOR formatted token, wrapped with a
+COSE_Sign1 <<COSE>> envelope. It is signed by the Platform attestation key and
+must contain a hash of the TVM public key, in order to create a cryptographic
```

Comment:
I am not sure this is needed, or even possible.  IIUC, the TSM token describes the TSM layer and is created by the Platform which I don't think has any knowledge of the TVM (it'd be a layering violation), in particular about TVM's pk.

The cryptographic chaining is obtained _naturally_ using the DICE delegation logics:
```
       .---------------------------.
       | Platform measurements     |
       | Platform Attestation PK   o--.
       +---------------------------+  |
SC o-->| HRoT Att SK signature     |  |
       '---------------------------'  |
                                      |
       .---------------------------.  |
       | TSM measurements          |  |
    .--o TSM Attestation PK        |  |
    |  +---------------------------+  |
    |  | Platform Att SK signature |<-'
    |  '---------------------------'
    |  
    |  .---------------------------.
    |  | nonce                     |
    |  | TVM measurements          |
    |  +---------------------------+
    '->| TSM Att SK signature      |
       '---------------------------'
```

which has the advantage of making the life of the verifier easier :-)

Reply:
Yes, you are right. It does not have (and probably would be challenging implementation wise...) to be that complex. It's fixed now.